extensions/pyRevitTools.extension/checks/radar_check.py

```diff
@@ -0,0 +1,624 @@
+        test_score = 1
+    # __________________________________________________________Check Test Score
+    if test_score >= 2:
+        output.print_md('{} All Tests Passed.'.format(GOOD_STRG))
+        script.exit()
+    else:
+        output.print_md( 
+        '{} Distant objects detected, Proceeding with additional analysis'.format(BAD_STRG))
+
+def check_linked_elements(document=doc):
+        # ___________________________________________________Check CAD and RVT Links
+    print(divider)
+    output.print_md('# Checking the extents of the CAD and RVT links')
+    print(divider)
+    badcads = get_far_elements(doc, cad=True, rvt=False)
+    badrvts = get_far_elements(doc, cad=False, rvt=True)
+    counter = 0
+    limit = 5
+    if len(badcads) > 0 or len(badrvts) > 0:
+        for x in badcads:
+            print(output.linkify(x.Id)+"__" + 
+                str(x.Name) + '  ' + str(x.Category.Name))
+            if counter == limit:
+                break
+            counter += 1
+        counter = 0
+        for x in badrvts:
+            print(output.linkify(x.Id)+"__" + 
+                str(x.Name) + '  ' + str(x.Category.Name))
+            if counter == limit:
+                break
+            counter += 1
+    else:
+        output.print_md('{} All CAD and RVT Links are located less than {}'.format(GOOD_STRG, CRITERIA_STRG))
+        test_score = 2
+        print(divider)
+
+    hide_linked_elements(document, make_temp_view(document), True, True)
+    cleanbbox = get_temp_view_bbox(doc)
+    if check_bounding_box(cleanbbox, INTERNAL_ORIGIN, EXTENT_DISTANCE) == 0:
+        output.print_md('{} Distant objects are still being detected!'.format(BAD_STRG))
+        output.print_md('{} Further Analysis Required.'.format(WARN_STRG))
+    else:
+        output.print_md('{} All Objects are located less than {}'.format(GOOD_STRG, CRITERIA_STRG))
+        script.exit()
+    print(divider)
+    output.print_md('# Checking everything, It is going to take a while.')
+    output.print_md('# Please be patient.')
+
+def check_all_elements(document=doc):
+    """
+    Checks the extents of all elements in the model.
+    Args:
+        doc (Document): The current document.
+    Returns:
+        None
+        """
+    print(divider)
+
+    badelements = get_far_elements(document,False, False, True)                                          
+    counter = 0
+    limit = 10
+    if len(badelements) > 0:
+        if len(badelements) > limit:
+            output.print_md('{} Showing the first 10 objects'.format(WARN_STRG))
+            output.print_md('{} Manual investigation is required'.format(WARN_STRG))
+        output.print_md('{} Elements below are located more than {}'.format(BAD_STRG, CRITERIA_STRG))
+        for x in badelements:
+            print(output.linkify(x.Id)+ '  ' + 
+                    str(x.Name) + '  ' + str(x.Category.Name))
+            if counter == limit:
+                break
+            counter += 1
```

Comment:
since you don't need the rest of the elements, just keep the first `limit` elements, no need to keep checking if you reach the limit

```suggestion
            badelements = badelements[:limit]
        output.print_md('{} Elements below are located more than {}'.format(BAD_STRG, CRITERIA_STRG))
        for x in badelements:
            print(output.linkify(x.Id)+ '  ' + 
                    str(x.Name) + '  ' + str(x.Category.Name))
```

Reply:
such a beautiful trick